src/TypeSystem/Inferrer/InferredTypes.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * PackageFactory.ComponentEngine - Universal View Components for PHP
+ *   Copyright (C) 2022 Contributors of PackageFactory.ComponentEngine
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+declare(strict_types=1);
+
+namespace PackageFactory\ComponentEngine\TypeSystem\Inferrer;
+
+use PackageFactory\ComponentEngine\TypeSystem\TypeInterface;
+
+class InferredTypes
```

Comment:
```suggestion
class TypeMap
```

It might be sensible to move this class to a namespace `PackageFactory\ComponentEngine\TypeSystem\Utility` or so. It may have more uses than just this one.

Reply:
Still not sure ;) maybe another day ^^ It will fall all into place sometime...